x-pack/plugins/reporting/server/export_types/printable_pdf/lib/pdf/get_template.ts

```diff
@@ -29,6 +35,79 @@ export function getTemplate(
   const subheadingMarginBottom = 5;
   const subheadingHeight = subheadingFontSize * 1.5 + subheadingMarginTop + subheadingMarginBottom;
 
+  const getStyle = (): StyleDictionary => ({
+    heading: {
+      alignment: 'left',
+      fontSize: headingFontSize,
+      bold: true,
+      margin: [headingMarginTop, 0, headingMarginBottom, 0],
+    },
+    subheading: {
+      alignment: 'left',
+      fontSize: subheadingFontSize,
+      italics: true,
+      margin: [0, 0, subheadingMarginBottom, 20],
+    },
+    warning: {
+      color: '#f39c12', // same as @brand-warning in Kibana colors.less
+    },
+  });
+  const getHeader = (): ContentText => ({
+    margin: [pageMarginWidth, pageMarginTop / 4, pageMarginWidth, 0],
+    text: title,
+    font: getFont(title),
+    style: {
+      color: '#aaa',
+    },
+    fontSize: 10,
+    alignment: 'center',
+  });
+  const getFooter = (): DynamicContent => (currentPage: number, pageCount: number) => {
+    const logoPath = path.resolve(assetPath, 'img', 'logo-grey.png'); // Default Elastic Logo
+    return {
+      margin: [pageMarginWidth, pageMarginBottom / 4, pageMarginWidth, 0],
+      layout: REPORTING_TABLE_LAYOUT,
+      table: {
+        widths: [100, '*', 100],
+        body: [
+          [
+            {
+              fit: [100, 35],
+              image: logo || logoPath,
+            },
+            {
+              alignment: 'center',
+              text: i18n.translate('xpack.reporting.exportTypes.printablePdf.pagingDescription', {
+                defaultMessage: 'Page {currentPage} of {pageCount}',
+                values: { currentPage: currentPage.toString(), pageCount },
+              }),
+              style: {
+                color: '#aaa',
+              },
+            },
+            '',
+          ],
+          [
+            logo
+              ? {
+                  text: i18n.translate('xpack.reporting.exportTypes.printablePdf.logoDescription', {
+                    defaultMessage: 'Powered by Elastic',
+                  }),
+                  fontSize: 10,
+                  style: {
+                    color: '#aaa',
+                  },
+                  margin: [0, 2, 0, 0],
+                }
+              : '',
+            '',
+            '',
+          ],
+        ],
+      },
+    };
+  };
+
   return {
     // define page size
```

Comment:
This huge object literal got split up, by moving property values out to helper functions.